src/charm.py

```diff
@@ -519,46 +625,53 @@ def _on_relation_handler(self, event: RelationEvent) -> None:
         self._connect_mongodb_exporter()
         self._connect_pbm_agent()
 
+        if (
+            self.db_initialised
+            and self.is_role(Config.Role.REPLICATION)
+            and self.lock_hash != self.databag_lock_hash
+        ):
+            # We write the lock file if it wasn't written yet.
+            self.lock_hash = self.databag_lock_hash
+
```

Comment:
now that we have several operations that need to be done by all-units in this handler , I think it might make sense to add a dedicated function for them